docs/create_new_backend.rst

```diff
@@ -0,0 +1,483 @@
+                                           include/onemkl/blas/detail/newlib/onemkl_blas_newlib.hpp \ # Output header file
+                                           onemkl::newlib                                             # Wrappers namespace
+
+Code snippet of the generated header file ``include/onemkl/blas/detail/newlib/onemkl_blas_newlib.hpp``
+
+.. code-block:: cpp
+
+    namespace onemkl {
+    namespace newlib {
+    
+    void asum(cl::sycl::queue &queue, std::int64_t n, cl::sycl::buffer<float, 1> &x, std::int64_t incx,
+              cl::sycl::buffer<float, 1> &result);
+
+
+
+**Compile-time Dispatching Interface Example**: command to generate the compile-time dispatching interface template instantiations for ``newlib`` and supported device ``newdevice``
+
+.. code-block:: bash
+
+    python scripts/generate_ct_instant.py   include/onemkl/blas/detail/blas_ct_templates.hpp \         # Base header file
+                                            include/onemkl/blas/detail/newlib/blas_ct.hpp \            # Output header file
+                                            include/onemkl/blas/detail/newlib/onemkl_blas_newlib.hpp \ # Header file with declaration of entry points to wrappers
+                                            newlib \                                                   # Library name
+                                            newdevice \                                                # Backend name
+                                            onemkl::newlib                                             # Wrappers namespace
+
+Code snippet of the generated header file ``include/onemkl/blas/detail/newlib/blas_ct.hpp``
+
+.. code-block:: cpp
+
+    namespace onemkl {
+    namespace blas {
+    
+    template <>
+    void asum<library::newlib, backend::newdevice>(cl::sycl::queue &queue, std::int64_t n,
+                                                   cl::sycl::buffer<float, 1> &x, std::int64_t incx,
+                                                   cl::sycl::buffer<float, 1> &result) {
+        asum_precondition(queue, n, x, incx, result);
+        onemkl::newlib::asum(queue, n, x, incx, result);
+        asum_postcondition(queue, n, x, incx, result);
+    }
+
+
+.. _integrate_header_files:
+
+2. Integrate Header Files
+-------------------------
+
+Below you can see oneMKL Header-based Part Structure:
+
+::
+
+    include/
+        onemkl/
+            onemkl.hpp -> oneMKL spec APIs
+            types.hpp  -> oneMKL spec types
+            detail/    -> implementation specific header files
+                exceptions.hpp        -> oneMKL exception classes
+                libraries.hpp         -> list of oneMKL libraries
+                backends.hpp          -> list of oneMKL backends
+                backends_selector.hpp -> oneMKL runtime dispatcher based on queue
+            blas/
+                predicates.hpp -> oneMKL BLAS pre-√ post-√
```

Comment:
I'd write out "pre-check", "post-check", etc.

Reply:
Agree, I will update it